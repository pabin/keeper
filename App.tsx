import React, { useState, useEffect } from 'react';
import { useColorScheme } from 'react-native';
import Toast from 'react-native-toast-message';

import { getObjectData } from './src/model/storageUtils';
import {
  createOrUpdateNoteDetails,
  updateNoteStatus,
} from './src/model/noteModel';
import StackNavigator from './src/navigation/StackNavigation';
import { themes } from './src/styles/colors';

export const NoteContext = React.createContext([]);

const App = () => {
  const [loading, setLoading] = useState(true);
  const [notes, setNotes] = useState([]);
  const [favouriteNotes, setFavouriteNotes] = useState([]);
  const [archivedNotes, setArchivedNotes] = useState([]);

  // To Do: Theme switch
  const isDarkMode = useColorScheme() === 'dark';

  const getDatabase = async () => {
    const notesData = await getObjectData('notes');

    if (notesData) {
      const favourites = notesData.filter(n => n.isFavourite);
      const archived = notesData.filter(n => n.isArchived);
      setNotes(notesData);
      setArchivedNotes(archived);
      setFavouriteNotes(favourites);
      setLoading(false);
    }
  };

  useEffect(() => {
    getDatabase();
  }, []);

  const onNoteStatusUpdate = async (note, action) => {
    await updateNoteStatus(note, action);
    await getDatabase();
  };

  const onCreateOrUpdateNote = async (note, title, body) => {
    await createOrUpdateNoteDetails(note, title, body);
    await getDatabase();
  };

  console.log('notes', JSON.stringify(notes));


  return (
    <>
      <NoteContext.Provider
        value={{
          notes,
          loading,
          favouriteNotes,
          archivedNotes,
          onNoteStatusUpdate,
          onCreateOrUpdateNote,
          theme: isDarkMode ? themes.dark : themes.light,
        }}
      >
        <StackNavigator />
        <Toast />
      </NoteContext.Provider>
    </>
  );
};

export default App;
