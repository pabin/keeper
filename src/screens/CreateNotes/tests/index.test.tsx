import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react-native';

import { note } from '../../../fixtures/notes';
import { NoteContext } from '../../../../App';
import CreateNoteScreen from '..';

describe('CreateNoteScreen', () => {
  afterEach(cleanup);

  const navigateMock = jest.fn();
  const navigation = {
    navigate: navigateMock,
  };

  const onCreateOrUpdateNote = jest.fn();

  const renderComponent = note => {
    const route = {
      params: { note },
    };
    return render(
      <NoteContext.Provider value={{ onCreateOrUpdateNote }}>
        <CreateNoteScreen route={route} navigation={navigation} />
      </NoteContext.Provider>,
    );
  };

  test('renders correctly in edit mode', () => {
    const { toJSON } = renderComponent(note);

    expect(toJSON()).toMatchSnapshot();
  });

  test('renders correctly in create mode', () => {
    const { toJSON } = renderComponent(undefined);

    expect(toJSON()).toMatchSnapshot();
  });

  test('updates note correctly', () => {
    const newNoteTitle = 'Updated note title';

    const { getByTestId } = renderComponent(note);

    const noteTitle = getByTestId('noteTitle');
    const createNote = getByTestId('createNote');

    fireEvent.changeText(noteTitle, newNoteTitle);
    fireEvent.press(createNote);

    expect(onCreateOrUpdateNote).toHaveBeenCalledWith(
      note,
      newNoteTitle,
      note.body,
    );
    expect(navigateMock).toHaveBeenCalledWith('Notes');
  });

  // test('creates note correctly', () => {
  //   const newNoteTitle = 'New note title';

  //   const { getByTestId } = renderComponent(undefined);

  //   const noteTitle = getByTestId('noteTitle');
  //   const createNote = getByTestId('createNote');

  //   fireEvent.changeText(noteTitle, newNoteTitle);
  //   fireEvent.press(createNote);

  //   expect(onCreateOrUpdateNote).toHaveBeenCalledWith(note, newNoteTitle, '');
  //   expect(navigateMock).toHaveBeenCalledWith('Notes');
  // });
});
