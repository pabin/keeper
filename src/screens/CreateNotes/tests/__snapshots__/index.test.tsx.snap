// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNoteScreen renders correctly in create mode 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 5,
          "paddingHorizontal": 5,
        }
      }
    >
      Title
    </Text>
    <TextInput
      maxLength={64}
      onChangeText={[Function]}
      placeholder="Note Title"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#CCCCCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 16,
          "height": 50,
          "paddingHorizontal": 10,
        }
      }
      testID="noteTitle"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
        "minHeight": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 5,
          "paddingHorizontal": 5,
        }
      }
    >
      Body
    </Text>
  </View>
  <View
    accessibilityLabel="Learn more"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="createNote"
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#01579B",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Create Note
      </Text>
    </View>
  </View>
</View>
`;

exports[`CreateNoteScreen renders correctly in edit mode 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 5,
          "paddingHorizontal": 5,
        }
      }
    >
      Title
    </Text>
    <TextInput
      maxLength={64}
      onChangeText={[Function]}
      placeholder="Note Title"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#CCCCCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 16,
          "height": 50,
          "paddingHorizontal": 10,
        }
      }
      testID="noteTitle"
      value="note title"
    />
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
        "minHeight": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 5,
          "paddingHorizontal": 5,
        }
      }
    >
      Body
    </Text>
  </View>
  <View
    accessibilityLabel="Learn more"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="createNote"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#01579B",
            },
          ]
        }
      >
        Update Note
      </Text>
    </View>
  </View>
</View>
`;
